bin/cron/delete_accounts

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/db'
+require 'cdo/only_one'
+
+OPEN_ENDED_LEVEL_TYPES = %w(
+  Applab
+  FreeResponse
+  Gamelab
+  Weblab
+).freeze
+
+# Deletes all projects and project data associated with a user.
+# @param [Integer] user_id for which to delete projects and project data.
+def delete_projects_and_project_data(user_id)
+  # TODO(asher): Delete channel-backed projects.
+
+  GalleryActivity.where(user_id: user_id).delete_all
+end
+
+# @param [Integer] user_id for which to delete free response answers.
+def delete_open_ended_answers(user_id)
+  UserLevels.where(user_id: user_id).find_each do |user_level|
+    if OPEN_ENDED_LEVEL_TYPES.include? user_level.level.type
+      user_level.level_source.destroy
+    end
+  end
+
+  Activity.where(user_id: user_id).find_each do |activity|
+    if OPEN_ENDED_LEVEL_TYPES.include? activity.level.type
+      activity.level_source.destroy
+    end
+  end
+end
+
+# @param [Integer] user_id for which to delete survey responses.
+def delete_survey_responses(user_id)
+  SurveyResult.where(user_id: user_id).destroy_all
+end
+
+# @param [Integer] user_id for which to delete orphaned students.
+def delete_orphaned_students(user_id)
+  section_ids = Section.with_deleted.
+    where(user_id: user.id).
+    where(login_type: [Section::LOGIN_TYPE_PICTURE, Section::LOGIN_TYPE_WORD]).
+    pluck(:id)
+  Follower.with_deleted.where(section_id: section_ids).find_each do |follower|
+    next unless student_user = follower.student_user
+    next unless student_user.provider == User::PROVIDER_SPONSORED
+    next unless Follower.where(student_user: student_user).count == 1
+
+    # TODO(asher): Determine whether this should be delete_user_and_dependencies(student_user).
+    student_user.destroy!
```

Comment:
Because we are fine with maintaining a record that the student did work, we want to eliminate the work that was done. This is motivated by student privacy law, I believe.